Can not install any plugins via helm

### Describe the bug

When I use a customized Helm responds file and enable the `controller.installPlugins` using the latest version of Jenkins JDK11 (2.332-jdk11) Docker Image, I got problem to download any plugin because of using `https://` protocol to `updates.jenkins.io/experimental/update-center.json` 

Error I got in the initialize stage of the deployment is. 

```
# kubectl logs jenkins-0 -c init
disable Setup Wizard
download plugins
File containing list of plugins to be downloaded: /var/jenkins_home/plugins.txt
Reading in plugins from /var/jenkins_home/plugins.txt

No directory to download plugins entered. Will use default of /usr/share/jenkins/ref/plugins
Using update center https://updates.jenkins.io/update-center.json from JENKINS_UC environment variable
Using experimental update center https://updates.jenkins.io/experimental/update-center.json from JENKINS_UC_EXPERIMENTAL environment variable
Using incrementals mirror https://repo.jenkins-ci.org/incrementals from JENKINS_INCREMENTALS_REPO_MIRROR environment variable
No CLI option or environment variable set for plugin info, using default of https://updates.jenkins.io/plugin-versions.json
Will use war file: /usr/share/jenkins/jenkins.war

Retrieving update center information
Update center URL: https://updates.jenkins.io/update-center.json?version=2.332
Cache miss for: update-center-2.332
io.jenkins.tools.pluginmanager.impl.UpdateCenterInfoRetrievalException: Error getting update center json
	at io.jenkins.tools.pluginmanager.impl.PluginManager.getJson(PluginManager.java:792)
	at io.jenkins.tools.pluginmanager.impl.PluginManager.getUCJson(PluginManager.java:814)
	at io.jenkins.tools.pluginmanager.impl.PluginManager.start(PluginManager.java:208)
	at io.jenkins.tools.pluginmanager.impl.PluginManager.start(PluginManager.java:172)
	at io.jenkins.tools.pluginmanager.cli.Main.main(Main.java:70)
Caused by: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at java.base/sun.security.ssl.Alert.createSSLException(Alert.java:131)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:349)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:292)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:287)
	at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.checkServerCerts(CertificateMessage.java:654)
	at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.onCertificate(CertificateMessage.java:473)
	at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.consume(CertificateMessage.java:369)
	at java.base/sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:392)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:443)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:421)
	at java.base/sun.security.ssl.TransportContext.dispatch(TransportContext.java:182)
	at java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:172)
	at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1426)
	at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1336)
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:450)
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:421)
	at java.base/sun.net.www.protocol.https.HttpsClient.afterConnect(HttpsClient.java:572)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:197)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1592)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1520)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:250)
	at java.base/java.net.URL.openStream(URL.java:1165)
	at org.apache.commons.io.IOUtils.toString(IOUtils.java:2953)
	at io.jenkins.tools.pluginmanager.impl.PluginManager.getJson(PluginManager.java:786)
	... 4 more
Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:439)
	at java.base/sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:306)
	at java.base/sun.security.validator.Validator.validate(Validator.java:264)
	at java.base/sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:313)
	at java.base/sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:222)
	at java.base/sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:129)
	at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.checkServerCerts(CertificateMessage.java:638)
	... 23 more
Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at java.base/sun.security.provider.certpath.SunCertPathBuilder.build(SunCertPathBuilder.java:141)
	at java.base/sun.security.provider.certpath.SunCertPathBuilder.engineBuild(SunCertPathBuilder.java:126)
	at java.base/java.security.cert.CertPathBuilder.build(CertPathBuilder.java:297)
	at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:434)
	... 29 more
Error getting update center json
```

If I disable the `controller.installPlugin` section of the yaml file, the installation of Jenkins Continues as expected and I need to download all plugins manually and using the non-secure protocol `http://`



### Version of Helm and Kubernetes

```markdown
- Helm: 
version.BuildInfo{Version:"v3.7.1", GitCommit:"1d11fcb5d3f3bf00dbe6fe31b8412839a96b3dc4", GitTreeState:"clean", GoVersion:"go1.16.9"}
- Kubernetes: 
Client Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.18-dispatcher", GitCommit:"de944d802c49735ad0f4bfe82ddfa19737ebe962", GitTreeState:"clean", BuildDate:"2021-05-13T17:51:55Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.3+k3s1", GitCommit:"61a2aab25eeb97c26fa3f2b177e4355a7654c991", GitTreeState:"clean", BuildDate:"2021-11-04T00:25:07Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/arm64"}
```


### Chart version

jenkins-3.10.2

### What happened?

```bash
1. Create a custom values.yaml file based on the original that exist in the this git repo.
2. Enable `controller.installPlugins`, adding a any plugin like git or kubernetes.  
3. Test to deploy your yaml file `helm install jenkins jenkins/jenkins -f config/values.yaml`
4. Wait and view the logs for the Init stage. `kubectl logs jenkins-0 -c init`
```


### What you expected to happen?

I expected the Jenkins Docker Container to be deployed and installing my plugins. 
Or I want to have a new parameter in the Values.yaml file to set a new patch for JENKINS_UC or be able to add the new certificate so we can use HTTPS.


### How to reproduce it

```markdown
controller:
  componentName: "jenkins-controller"
  image: "jenkins/jenkins"
  tag: "2.332-jdk11"
  installPlugins:
    - kubernetes:1.31.3
  installLatestPlugins: false
```


### Anything else we need to know?

_No response_

I have found the 
```
initContainerEnv:
    - name: JENKINS_UC
      value: "http://updates.jenkins.io/update-center.json"
 ```
Function but that doesn't help. It does only modify the error.
```
Retrieving update center information
Update center URL: http://updates.jenkins.io/update-center.json?version=2.319.2
Cache miss for: update-center-2.319.2
io.jenkins.tools.pluginmanager.impl.UpdateCenterInfoRetrievalException: Error getting update center json
	at io.jenkins.tools.pluginmanager.impl.PluginManager.getJson(PluginManager.java:792)
	at io.jenkins.tools.pluginmanager.impl.PluginManager.getUCJson(PluginManager.java:814)
	at io.jenkins.tools.pluginmanager.impl.PluginManager.start(PluginManager.java:208)
	at io.jenkins.tools.pluginmanager.impl.PluginManager.start(PluginManager.java:172)
	at io.jenkins.tools.pluginmanager.cli.Main.main(Main.java:70)
Caused by: java.io.FileNotFoundException: http://updates.jenkins.io/update-center.json?version=2.319.2
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1920)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1520)
	at java.base/java.net.URL.openStream(URL.java:1165)
	at org.apache.commons.io.IOUtils.toString(IOUtils.java:2953)
	at io.jenkins.tools.pluginmanager.impl.PluginManager.getJson(PluginManager.java:786)
	... 4 more
Error getting update center json
```

In the original error I'm seeing:
> Caused by: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

Which says to me that the init container was unable to verify the certificate from updates.jenkins.io. I've seen this happen when behind a VPN or other network security proxy that requires clients to trust a non-standard certificate authority. When using an `http` URL instead, that's why you see a different error.

@Issen007 can you confirm you're running behind a proxy/VPN?

You can configure an http proxy following the [documentation](https://github.com/jenkinsci/helm-charts/tree/main/charts/jenkins#running-behind-a-forward-proxy), but you'll have to work with your IT team to setup an http proxy that will work correctly.

--- a/charts/jenkins/CHANGELOG.md
+++ b/charts/jenkins/CHANGELOG.md
@@ -12,6 +12,10 @@ Use the following links to reference issues, PRs, and commits prior to v2.6.0.
 The change log until v1.5.7 was auto-generated based on git commits.
 Those entries include a reference to the git commit to be able to get more details.
 
+## 4.1.12
+
+If keystore is defined, it is now also made available in the initContainer.
+
 ## 4.1.11
 
 JCasC ConfigMaps now generate their name from the `jenkins.casc.configName` helper

--- a/charts/jenkins/Chart.yaml
+++ b/charts/jenkins/Chart.yaml
@@ -1,7 +1,7 @@
 apiVersion: v2
 name: jenkins
 home: https://jenkins.io/
-version: 4.1.11
+version: 4.1.12
 appVersion: 2.346.1
 description: Jenkins - Build great things at any scale! The leading open source automation server, Jenkins provides hundreds of plugins to support building, deploying and automating any project.
 sources:

--- a/charts/jenkins/templates/jenkins-controller-statefulset.yaml
+++ b/charts/jenkins/templates/jenkins-controller-statefulset.yaml
@@ -152,6 +152,11 @@ spec:
             - mountPath: {{ .Values.controller.jenkinsHome }}/init.groovy.d
               name: init-scripts
             {{- end }}
+            {{- if .Values.controller.httpsKeyStore.enable }}
+            {{- $httpsJKSDirPath :=  printf "%s" .Values.controller.httpsKeyStore.path }}
+            - mountPath: {{ $httpsJKSDirPath }}
+              name: jenkins-https-keystore
+            {{- end }}
       containers:
         - name: jenkins
           image: "{{ .Values.controller.image }}:{{- include "controller.tag" . -}}"

